fix(ast): restore `#[ts]` attribute to `FormalParameter` `decorators` field

overlookmotel opened the pull request:
Revert #11494.

As discussed in https://github.com/oxc-project/oxc/issues/11485#issuecomment-2943276060, whether the `#[ts]` attribute is present or not does not affect the code for generating these nodes in ESTree AST, because `FormalParameter` has a custom serializer and custom type def.

So the purpose of the `#[ts]` attribute here is more of a comment that this field is not present in the plain JS ESTree AST. That field is not present, so it's preferable to keep the attribute to signal that.

Boshen commented:
### Merge activity

* **Jun 5, 10:23 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(ast): restore `#[ts]` attribute to `FormalParameter` `decorators` field (#11497)

Revert #11494.

As discussed in https://github.com/oxc-project/oxc/issues/11485#issuecomment-2943276060, whether the `#[ts]` attribute is present or not does not affect the code for generating these nodes in ESTree AST, because `FormalParameter` has a custom serializer and custom type def.

So the purpose of the `#[ts]` attribute here is more of a comment that this field is not present in the plain JS ESTree AST. That field is not present, so it's preferable to keep the attribute to signal that.
